packages/crypto/src/transactions/builders/transactions/multi-payment.ts

```diff
@@ -19,8 +20,10 @@ export class MultiPaymentBuilder extends TransactionBuilder<MultiPaymentBuilder>
     }
 
     public addPayment(recipientId: string, amount: string): MultiPaymentBuilder {
-        if (this.data.asset.payments.length >= 500) {
-            throw new MaximumPaymentCountExceededError(this.data.asset.payments.length + 1);
+        const limit: number = configManager.getMilestone().multiPaymentLimit || 500;
+
+        if (this.data.asset.payments.length >= limit) {
+            throw new MaximumPaymentCountExceededError(limit, this.data.asset.payments.length + 1);
```

Comment:
This should just be `payments.length > limit` and the `+1` should be removed. It was previously already wrong since `limit` is inclusive.

Reply:
Fixed!

Reply:
For that to work the transaction would need to be pushed to the array before checking the length actually 🤔 I will revert the last commit, #3133 will refactor this specific error anyway.